drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -78,23 +71,14 @@ static struct nrf5_802154_data nrf5_data;
 #if defined(CONFIG_IEEE802154_NRF5_UICR_EUI64_ENABLE)
 #if defined(CONFIG_SOC_NRF5340_CPUAPP)
 #if defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
-#define EUI64_ADDR (NRF_UICR_S->OTP)
+#error "NRF_UICR->OTP is not supported to read from non-secure"
```

Comment:
@czeslawmakarski From what I can tell this has never worked, is that correct?
Neither SPM or TF-M has support for reading UICR at the moment.

Reply:
Yes, you are right. @e-rk Could you back me on that?

Reply:
I am not sure. I have never built this file with trusted firmware. It would probably be better to ask @Damian-Nordic if they ever used this configuration.

Reply:
I was fixing this code to support TF-M in OpenThread samples, but I have tested the default FICR case only, so I can't tell if the UICR has ever worked.

Reply:
Only FICR is supported by TF-M and SPM. So if this is a use-case it should be raised as a feature request.
Atleast now the code is correctly throwing an error.